From 7b7adad8b6fe2420b0574555a7ab483205bbca57 Mon Sep 17 00:00:00 2001
From: Peter Swain <pswain@cavium.com>
Date: Thu, 24 Jul 2014 18:05:50 -0700
Subject: [PATCH 734/974] MIPS: octeon-i2c: export twsix symbols for usb
 modules

Without these cvmx_twsi_xxx exported, CONFIG_USB_XHCI_HCD_OCTEON=m fails.
Also needed for other hardware (eg TDM/SLIC).
But perhaps all of these should use genric linux i2c API?
[Original patch taken from Cavium SDK 3.1.2-568]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 arch/mips/cavium-octeon/executive/cvmx-twsi.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/mips/cavium-octeon/executive/cvmx-twsi.c b/arch/mips/cavium-octeon/executive/cvmx-twsi.c
index 81aaefe..d750e4d 100644
--- a/arch/mips/cavium-octeon/executive/cvmx-twsi.c
+++ b/arch/mips/cavium-octeon/executive/cvmx-twsi.c
@@ -412,6 +412,7 @@ retry:
 	return num_bytes;
 #endif
 }
+EXPORT_SYMBOL(cvmx_twsix_read);
 
 /**
  * Perform a twsi write operation to a 7 bit device address.
@@ -507,6 +508,7 @@ retry:
 	return 0;
 #endif
 }
+EXPORT_SYMBOL(cvmx_twsix_write);
 
 /**
  * Write 1-8 bytes to a TWSI device using an internal address.
@@ -678,6 +680,7 @@ retry:
 	return num_bytes;
 #endif
 }
+EXPORT_SYMBOL(cvmx_twsix_write_ia);
 
 #ifndef CVMX_BUILD_FOR_LINUX_KERNEL
 /* Controller command patterns */
-- 
2.6.2

